OpenSim/Common/DataTable.h

```diff
@@ -341,38 +485,151 @@ class DataTable_ : public AbstractDataTable {
         return _indData;
     }
 
-    /** Get dependent column.                                                 
+    /** Get dependent column at index.
 
-    \throws ColumnIndexOutOfRange If index is out of range.                   */
+    \throws ColumnIndexOutOfRange If index is out of range for number of columns
+                                  in the table.                               */
     VectorView getDependentColumnAtIndex(size_t index) const {
         OPENSIM_THROW_IF(isColumnIndexOutOfRange(index),
                          ColumnIndexOutOfRange, index, 0,
-                         static_cast<size_t>(_depData.ncol()));
-        return _depData.col((int)index);
+                         static_cast<size_t>(_depData.ncol() - 1));
+
+        return _depData.col(static_cast<int>(index));
     }
 
-    /** Get dependent Column which has the given column label.                */
-    VectorView getDependentColumn(const std::string& columnLabel) {
+    /** Get dependent Column which has the given column label.                
+
+    \throws KeyNotFound If columnLabel is not found to be label of any existing
+                        column.                                               */
+    VectorView getDependentColumn(const std::string& columnLabel) const {
         return _depData.col(static_cast<int>(getColumnIndex(columnLabel)));
     }
 
-    /** Set independent column at index.                                      
+    /** Update dependent column at index.
+
+    \throws ColumnIndexOutOfRange If index is out of range for number of columns
+                                  in the table.                               */
+    VectorView updDependentColumnAtIndex(size_t index) {
+        OPENSIM_THROW_IF(isColumnIndexOutOfRange(index),
+                         ColumnIndexOutOfRange, index, 0,
+                         static_cast<size_t>(_depData.ncol() - 1));
+
+        return _depData.updCol(static_cast<int>(index));
+    }
+
+    /** Update dependent Column which has the given column label.
+
+    \throws KeyNotFound If columnLabel is not found to be label of any existing
+                        column.                                               */
+    VectorView updDependentColumn(const std::string& columnLabel) {
+        return _depData.updCol(static_cast<int>(getColumnIndex(columnLabel)));
+    }
+
+    /** %Set independent column at index.                                      
 
     \throws RowIndexOutOfRange If index is out of range.                        
     \throws InvalidRow If this operation invalidates the row. Validation is
                        performed by derived classes.                          */
     void setIndependentColumnAtIndex(size_t index, const ETX& value) {
```

Comment:
Maybe `index` should be named `rowIndex`. This is something that is generally confusing to me across the interface (when is the index a rowIndex or a colIndex?). I know that sometimes it only makes sense to be one or the other, but it takes some additional mental effort to realize that.